### DISCUSSION TITLE
What is your image optimization workflow?

### DISCUSSION THREAD

**Discussion by dodyg** [OrchardCMS/OrchardCore#7058] [Q&A]
OrchardCore has a very handy web resizing utility that just work by appending ?width and ?height on any images. This is unfortunately not enough for most website because even resized images can be optimized further to reduce their size. 

I am wondering if anyone has implemented automatic image optimization on the fly with their orchard core installation.

**Comment by Skrypt** [comment]
I believe this is a setting that can be set on ImageSharp compression ratio. Used to be. Like you could say I want my .jpg to be compressed up to 70% to reduce size / quality of the images. Though, from a web graphist point of view compressing too much is not always a good thing.

**Reply by dodyg** [reply]
I don't know much about image optimization but is it really just a matter of compression level especially for png?

**Comment by hishamco** [comment]
> ImageSharp compression ratio

There's a `CompressionLevel` property for the encoder option that you can set as a compression rate

**Comment by dodyg** [comment]
I checked with the library owner 
https://github.com/SixLabors/ImageSharp.Web/discussions/120

**Reply by Skrypt** [reply]
Depends always on the algorithm used to compress those images. Some might be better than others.
So, good thing to know that ImageSharp will allow to use external compression libs.

https://tinyjpg.com/developers/reference/dotnet

@dodyg Could your confirm that after using this image compressor you are getting optimized results in website analyzers?
We might make it an option in OC or look into ImageSharp to investigate further and see if it can achieve the same results.

**Reply by dodyg** [reply]
I am using tinyjpg.com manually and they do produce amazing results most of the time (60% - 70%) reduction in size.

Yeah image cropping won't cut it anymore for production website without these type image optimization because you still can end up with large size files. This is especially problematic when people without graphic skills upload "nice" pictures. A website can easily balloon to beyond 1MB.

**Reply by dodyg** [reply]
ImageSharp.Web just acquired the ability to to set JPEG image quality https://github.com/SixLabors/ImageSharp.Web/pull/125

**Reply by dodyg** [reply]
OrchardCore now supports image quality for JPG (+ converting an image to specific format) https://github.com/OrchardCMS/OrchardCore/commit/f7a1d95521385fd8e87b621f87c2c24ca2ceb770
